proposals/oci-support-updates.md

```diff
@@ -0,0 +1,95 @@
+# Proposal: OCI Support Updates
+
+A couple of the maintainers recently met regarding how to move forward with Helm's experimental OCI support. This documents the conclusions reached.
+
+Note: the existing support for OCI registries is documented [here](https://helm.sh/docs/topics/registries/).
+
+## 1. Implement `Getter` and introduce `Pusher`
+
+As of yet, the code related to registry support has been written standalone (see `internal/experimental/registry`).
+
+The act of downloading a chart from an OCI registry should mimic what is already possible using downloader plugins. In theory, using a protocol prefix such as `oci://` in any place where chart repos are referenced should work just by implementing a `Getter` for OCI.
+
+Currently there is no Helm-specific way to upload chart packages. Using the same model of `Getter`, a new interface called `Pusher` should be introduced, with a single built-in implementation (OCI). This opens the door for plugins to implement their own upload mechanism, and for Helm to add a new top-level upload command such as `helm push`.
+
+
+## 2. Support for provenance files
+
+Helm currently has the ability to verify package signatures, assuming the presence of a file suffixed with `.prov` sitting next to a chart `.tgz` in a repository.
+
+Support for this method of signature validation should be carried over into OCI storage. As the format of the OCI manifest is custom to Helm, Helm can also choose to modify the resulting manifest on upload when a chart is being signed (e.g. `helm push --sign`).
+
+As far as the low-level details, the `.prov` file will simply be stored as a second layer on the manifest. If running `helm pull --verify oci://...`, the second layer (layer at index 1) will be assumed to be the provenance file. The first layer is always the chart `.tgz` itself.
```

Comment:
What if we supported pushing the provenance file as the manifest.config (with a different mediaType from the current)? That would simplify access to it, without needing to assume anything, and (unless I'm mistaken) it's a superset of the Chart.yaml data used as the config now. Although the config is current in JSON (I guess for historical reasons), while the current .prov is a pair of YAML documents with an (undemarcated) signature appended to the second.

This would let a Helm-aware registry more-easily validate and display the presence and state of a signature in a browse UI, for example.

Side-note: I haven't checked, but do we actually validate the config when the chart is pulled?